tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -98,51 +98,28 @@ public Span? RootSpan
         /// </summary>
         internal string? AdditionalW3CTraceState { get; set; }
 
-        /// <summary>
-        /// Gets the IAST context.
-        /// </summary>
+        /// <summary> Gets the IAST context </summary>
         internal IastRequestContext? IastRequestContext => _iastRequestContext;
 
-        internal static TraceContext? GetTraceContext(in ArraySegment<Span> spans) =>
-            spans.Count > 0 ?
-                spans.Array![spans.Offset].Context.TraceContext :
-                null;
-
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private AppSecRequestContext GetRequestContext()
+        /// <summary> Gets the AppSec context </summary>
+        internal AppSecRequestContext AppSecRequestContext
```

Comment:
nit: I wonder if it's worth making this a method called `GetOrCreateAppSecRequestContext()`? I'm undecided if there's any value in that tbh, just a thought 🤷 

Mostly because the _subtle_ difference with `IastRequestContext` being _potentially_ null, while `AppSecRequestContext` is _never_ `null` slightly concerns me 🤔

Reply:
I thought the same, but I preferred the symmetry, and the nullable return does the job also, IMO